RELEASE/scripts/autoscend/auto_post_adv.ash

```diff
@@ -420,14 +420,26 @@ boolean auto_post_adventure()
 		{
 			use_skill(1, $skill[Soul Rotation]);
 		}
-		int missing = (my_maxmp() - my_mp()) / 15;		//soul food restores 15 MP per cast.
+		
+		int maxMPNextTurn;
+		int MPtoRestore = my_maxmp() - my_mp();
+		if(MPtoRestore >= 15 && my_maxmp() > 150 && my_soulsauce() < 90)	//todo how much MP wanted for buffing before next turn?
```

Comment:
What is purpose of soulsauce condition?

Reply:
probably efficiency but this check seems wrong, all 100 soulsauce could get wasted if big effects are expiring

Reply:
Efficiency from a computation or resource conservation point of view? Don't have my head fully wrapped around the logic but it seems that we shouldn't have the soulsauce check here. Regardless of how much we have, we shouldn't restore MP that we will be losing.

Also, as a more general question - looks like majority of our buffing is after this code. Restoring to max mp is only a waste if we don't use it all before the next turn (when +max MP effects expire). What are your thoughts on adding similar logic to wherever we restore MP for casting buffs? Only an issue when restoring for buffing if we restore just enough to cast buff or if we try to top off. I took a look and wasn't clear to me where restoring MP for buffing is located...

Reply:
this is also something I wondered about because some acquireMP restore is done from pre_adv but that can fail without even looking at soul food if the setting to ignore restore failures is default, it seems like the only restore is static 32 MP for combat in pre_adv probably for weaksauce + saucegeyser?
would want to know how much mp is wanted for buffing, then how much mp can be used for buffing, this depends on how much mp would be wanted for combat, and goes back to combat predictions

but the my_soulsauce condition here should go. I think it meant to avoid simulating effects for small changes that would be used for buffs but this can't be known yet

Reply:
#1128